sdk/typespec/typespec_client_core/src/tracing/attributes.rs

```diff
@@ -1,17 +1,134 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+/// An array of homogeneous attribute values.
+#[derive(Debug, PartialEq, Clone)]
 pub enum AttributeArray {
+    /// An array of boolean values.
     Bool(Vec<bool>),
+    /// An array of 64-bit signed integers.
     I64(Vec<i64>),
-    U64(Vec<u64>),
+    /// An array of 64bit floating point values.
+    F64(Vec<f64>),
+    /// An array of strings.
     String(Vec<String>),
 }
 
+/// Represents a single attribute value, which can be of various types
+#[derive(Debug, PartialEq, Clone)]
 pub enum AttributeValue {
+    /// A boolean attribute value.
     Bool(bool),
+    /// A signed 64-bit integer attribute value.
     I64(i64),
-    U64(u64),
+    /// A 64-bit floating point attribute value
+    F64(f64),
+    /// A string attribute value.
     String(String),
+    /// An array of attribute values.
     Array(AttributeArray),
 }
+
+/// Represents a key-value pair attribute, which is used for tracing and telemetry.
+///
+/// Attributes are used to provide additional context and metadata about a span or event.
+/// They can be of various types, including strings, integers, booleans, and arrays.
+///
+/// Attributes are typically used to enrich telemetry data with additional information
+/// that can be useful for debugging, monitoring, and analysis.
+#[derive(Debug, PartialEq, Clone)]
+pub struct Attribute {
+    /// A key-value pair attribute.
+    pub key: &'static str,
```

Comment:
There are well-defined attributes for which the key is computed at runtime. Most database conventions, [Cosmos' included](https://opentelemetry.io/docs/specs/semconv/database/cosmosdb/) define `db.query.parameter.<key>` for encoding the values of query parameters.

If we switch this to a `Cow<'static, str>` we should largely get the best of both worlds. We'd be able to use static strings without heap allocation but still support heap-allocated dynamic strings where needed.